src/App.tsx

```diff
@@ -1,44 +1,112 @@
+  const [selectedUserId, setSelectedUserId] = useState<number>(-1);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedPostId, setSelectedPostId] = useState<number | null>(null);
+
+  const loadDataUsers = async () => {
+    try {
+      const usersData = await getUsers();
+
+      setUsers(usersData);
+    } catch {
+      setIsError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadDataUsers();
+  }, []);
+
+  const selectedUserHandleChange = useCallback((value: number) => {
+    setSelectedUserId(value);
+  }, [selectedUserId]);
+
+  const loadDataUserPosts = async () => {
+    try {
+      setIsLoading(true);
+      const postsData = await getUserPosts(selectedUserId);
+
+      setIsLoading(false);
+      setUserPosts(postsData);
+    } catch (error) {
+      setIsError(true);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadDataUserPosts();
+  }, [selectedUserId]);
+
+  const selectedPostHandleChange = useCallback((value: number | null) => {
+    setSelectedPostId(value);
+  }, [selectedPostId]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectUserId={selectedUserHandleChange}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {selectedUserId < 0 && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {selectedUserId !== -1 && !userPosts.length && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
 
-                <PostsList />
+                {!!userPosts.length && (
```

Comment:
```suggestion
                {!!userPosts.length && !isLoading (
```
I see posts of the previous user during the loading
the same with `No posts yet` message

Reply:
the same with comments